src/Components/ProfileListPage.js

```diff
@@ -149,7 +156,16 @@ export default function ProfileListPage() {
           </Tooltip>
         )}
       </Box>
-
+      {listId.toLowerCase() !== "likes" &&
+        listId.toLowerCase() !== "dislikes" && (
+          <ClickAndEdit
+            data={desc}
+            canEdit={canEdit}
+            update={updateListDesc}
+            index={index}
```

Comment:
If you want to not need the index dependency here, you could write ClickAndEdit to just take an `onSave` function that has a single string arg with the new value the user is saving.  Then the logic for providing the index in the call to `updateListDesc` could be extracted out here, and the ClickAndEdit component wouldn't have to have any knowledge of indices, and wouldn't really care if an edited value was part of an array or not, since any logic like that would happen outside it.

Reply:
Ex:

Above, 
```js
const onDescSave = (newDesc, index) => updateListDesc(newDesc, index);
```

Then, when creating a ClickAndEdit:
```js
<ClickAndEdit data={desc} canEdit={canEdit} onSave={(newDesc) => onDescSave(newDesc, index)}/>
```

Reply:
Epic hack!! 10/10.